### PR TITLE
add min/max values for all selections

### PR DESCRIPTION
**Pull request by rvanharen**

### CODE REVIEW

File: c3s_magic_wps/processes/wps_drought_indicator.py [hunk 1]
```diff
@@ -32,6 +33,8 @@ def __init__(self):
                          default='ERA-Interim',
                          min_occurs=1,
                          max_occurs=1),
+            *reference_year_ranges((1980, 1989)),
```
Comment (nielsdrost) [inline c3s_magic_wps/processes/wps_drought_indicator.py]: This needs the recipe template to also know about reference years, and code to pass it to the template

File: c3s_magic_wps/processes/wps_extreme_events.py [hunk 2]
```diff
@@ -24,7 +25,9 @@ def __init__(self):
                          default='ERA-Interim',
                          min_occurs=1,
                          max_occurs=1),
+            *reference_year_ranges((1980, 1989)),
```
Comment (nielsdrost) [inline c3s_magic_wps/processes/wps_extreme_events.py]: This needs the recipe template to also know about reference years, and code to pass it to the template

File: c3s_magic_wps/processes/wps_teleconnections.py [hunk 3]
```diff
@@ -32,6 +33,7 @@ def __init__(self):
                          default='ERA-Interim',
                          min_occurs=1,
                          max_occurs=1),
+            *reference_year_ranges((1980, 1989)),
```
Comment (nielsdrost) [inline c3s_magic_wps/processes/wps_teleconnections.py]: This needs the recipe template to also know about reference years, and code to pass it to the template

File: c3s_magic_wps/processes/wps_weather_regimes.py [hunk 4]
```diff
@@ -31,6 +32,7 @@ def __init__(self):
                          default='ERA-Interim',
                          min_occurs=1,
                          max_occurs=1),
+            *reference_year_ranges((1980, 1989)),
```
Comment (nielsdrost) [inline c3s_magic_wps/processes/wps_weather_regimes.py]: This needs the recipe template to also know about reference years, and code to pass it to the template

**Comment by nielsdrost** [comment]
Thanks @rvanharen! I've marked a few lines where some additional fixes are needed to make the new settings also have an effect :-)

File: c3s_magic_wps/processes/wps_teleconnections.py (hunk 3 above)
Comment (rvanharen) [reply c3s_magic_wps/processes/wps_teleconnections.py]: should be fixed by https://github.com/c3s-magic/c3s-magic-wps/pull/98/commits/c4c6190a878e233229b3be9b8a2a99c260abaf31

File: c3s_magic_wps/processes/wps_weather_regimes.py (hunk 4 above)
Comment (rvanharen) [reply c3s_magic_wps/processes/wps_weather_regimes.py]: https://github.com/c3s-magic/c3s-magic-wps/pull/98/commits/c4c6190a878e233229b3be9b8a2a99c260abaf31

File: c3s_magic_wps/processes/wps_drought_indicator.py (hunk 1 above)
Comment (rvanharen) [reply c3s_magic_wps/processes/wps_drought_indicator.py]: https://github.com/c3s-magic/c3s-magic-wps/pull/98/commits/c4c6190a878e233229b3be9b8a2a99c260abaf31

File: c3s_magic_wps/processes/wps_extreme_events.py (hunk 2 above)
Comment (rvanharen) [reply c3s_magic_wps/processes/wps_extreme_events.py]: https://github.com/c3s-magic/c3s-magic-wps/pull/98/commits/c4c6190a878e233229b3be9b8a2a99c260abaf31

**Comment by rvanharen** [comment]
Should be fixed now if I didn't miss any of them. Also we need to set both minval and maxval, else the pywps check fails according to travis, so the last commit should fix those as well if I caught them all.

**Pull request merged**